lib/settings.h

```diff
@@ -444,6 +444,13 @@ class CPPCHECKLIB WARN_UNUSED Settings {
         return jobs == 1;
     }
 
+    /** Check if the user wants to check for unused functions
+     * and if it's possible at all */
+    bool isUnusedFunctionCheckEnabled() const
+    {
+        return useSingleJob() && checks.isEnabled(Checks::unusedFunction);
```

Comment:
hmm.. I wonder why it doesn't check `useSingleJob() || !buildDir.empty()`

Reply:
Just keeping the current behavior intact for now.

Reply:
Unsurprisingly tis check turned out flawed. This is addressed in #6207.